doc/phar-signing.md

```diff
@@ -0,0 +1,364 @@
+
+When publishing your archive, you should publish both `bin/command.phar` and `bin/command.phar.asc`.
+
+
+### Verifying the PHAR signature
+
+First you should check the issuer's identity, usually it is provided from where you download it as part of the
+documentation:
+
+```
+gpg --keyserver hkps://keys.openpgp.org --recv-keys 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+```
+
+However not everyone expose what is their GPG key ID. So sometimes to avoid bad surprises, you
+can look up for similar issuers to the key ID given by the `.asc`:
+
+```
+# Verify the signature
+$ gpg --verify bin/command.phar.asc bin/command.phar
+
+# Example of output:
+gpg: Signature made Sat 21 Oct 16:58:05 2023 CEST
+gpg:                using EDDSA key 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+gpg: Good signature from "Théo Fidry <theo.fidry+phar-signing-example@example.com>" [ultimate]
+```
+
+If the key ID was not provided before, you can try to look it up to check it was properly registered to a keyserver:
+
+```
+$ gpg --keyserver https://keys.openpgp.org --search-keys "theo.fidry+phar-signing-example@example.com"
+```
+
+
+## Automatically sign in GitHub Actions
+
+The first step is to add [environment secrets to your repository][github-environment-secrets]:
+
+```
+$ gpg --export-secret-key --armor 96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+# Paste the content into a secret environment variable
+GPG_KEY_96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08
+
+# Add the corresponding passphrase enviroment variable:
+GPG_KEY_96C8013A3CC293C465EE3FBB03B2F4DF7A20DF08_PASSPHRASE
+```
+
+Then you need to:
+- Build your PHAR
+- Import the GPG key
+- Sign your PHAR
+- Publish your PHAR
+
+I highly recommend to build your PHAR as part of your regular workflows. Then the other steps can be enable on release
+only. The following is an example of GitHub workflow:
+
+```yaml
+name: Release
+
+on:
+    push:
+        branches: [ main ]
+    pull_request: ~
+    schedule:
+        # Do not make it the first of the month and/or midnight since it is a very busy time
+        - cron: "* 10 6 * *"
+    release:
+        types: [ created ]
+
+# See https://stackoverflow.com/a/72408109
+concurrency:
+    group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    build-phar:
+        runs-on: ubuntu-latest
+        name: Build PHAR
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v3
+                with:
+                    fetch-depth: 0
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: '8.1'
+                    ini-values: phar.readonly=0
+                    tools: composer
+                    coverage: none
+
+            -   name: Install Composer dependencies
+                uses: ramsey/composer-install@v2
+
+            -   name: Build PHAR
+                run: ...
```

Comment:
Do you think we should add support for gpg signing to box? We (as in the phar.io team) are currently discussing on extending our gpg library to do that. Would be nice to have out of the box support (no pun here ;-) )...

Reply:
100% Agree. I'll try to get our library in shape for things like this. And maybe - and that's not a promise yet ;-) - I'll try to get a PR ready to box to include that. But I have no real ETA for that as of yet..